From 1a883553a2ce87bee2609f4b2e783150636d7dee Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Mon, 14 Oct 2013 11:16:43 +0200
Subject: [PORT FROM MAIN] wifi: Fix issue when trying to connect to EAP-SIM alone with USIM inserted

BZ: 64516

When eap method is set to SIM only in selected network from wpa_supplicant conf
while a USIM is inserted, the connection don't start.

In fact, PCSC initialisation depends on selected methods:
  - when both SIM and AKA methods are selected, PCSC supports
    the both sim types (GSM SIM & USIM)
  - when AKA method is only selected, PCSC supports
    only USIM sim type.
  - when SIM method is only selected, PCSC supports
    only GSM sim type.

This selection of type depending of the method cause the initialisation
issue when a USIM is inserted with SIM only method.

Alway initializing PCSC with both type support avoid such issue.

Change-Id: I3c492ac97f70078bcb7c6a2e1a712f8ca36d20f9
Orig-Change-Id: I029761d5f051965d8f3ca5ad9f5b6501331501d3
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Honore Tricot <honorex.tricot@intel.com>
---
 wpa_supplicant/events.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/wpa_supplicant/events.c b/wpa_supplicant/events.c
index ec35a1c..562abb9 100644
--- a/wpa_supplicant/events.c
+++ b/wpa_supplicant/events.c
@@ -316,7 +316,6 @@ int wpa_supplicant_scard_init(struct wpa_supplicant *wpa_s,
 
 	wpa_dbg(wpa_s, MSG_DEBUG, "Selected network is configured to use SIM "
 		"(sim=%d aka=%d) - initialize PCSC", sim, aka);
-
 	wpa_s->scard = scard_init((!sim && aka) ? SCARD_USIM_ONLY :
 				  SCARD_TRY_BOTH, NULL);
 	if (wpa_s->scard == NULL) {
-- 
1.7.4.1

